hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -3905,8 +3905,32 @@ median' input = unsafePerformIO $ ATen.cast3 ATen.Managed.median_tlb
 -- argsort :: Tensor device dtype shape -> Int -> Bool -> Tensor device dtype shape
 -- argsort _input _dim _descending = unsafePerformIO $ (ATen.cast3 ATen.Managed.argsort_tlb) _input _dim _descending
 
--- topk :: Tensor device dtype shape -> Int -> Int -> Bool -> Bool -> (Tensor device dtype shape,Tensor device dtype shape)
--- topk _input _k _dim _largest _sorted = unsafePerformIO $ (ATen.cast5 ATen.Managed.topk_tllbb) _input _k _dim _largest _sorted
+type family TopKCheck (k :: Nat) (shape :: [Nat]) (dim :: Nat) (satd :: Maybe Nat) (result :: Maybe a) :: a where
+  TopKCheck _ shape dim _ Nothing       = DimOutOfBound shape dim
+  TopKCheck _ shape dim Nothing _       = DimOutOfBound shape dim
+  TopKCheck k shape dim (Just v) (Just result) = If ( k <=? v ) result (TypeError (Text "k must be less than or equal to the number of elements in the requested dimention and greater than or equal to 1"))
```

Comment:
awesome, this is even better! though, have you tried k=0? it should just work and return an empty tensor.
there’s a typo in the error message, “dimension”.

Reply:
fixed typo and now supports k = 0